fd/diff.go

```diff
@@ -19,25 +18,30 @@ type Point struct {
 	Coeff float64
 }
 
-// Method represents a finite difference formula that approximates
-// the derivative of order k of a function f at x as
-//  d^k f(x) ≅ (1 / h^k) * \sum_i Coeff_i * f(x + h * Loc_i),
-// where h is a small positive step.
-type Method struct {
+// Formula represents a finite difference formula on a regularly spaced grid
+// that approximates the derivative of order k of a function f at x as
+//  d^k f(x) ≅ (1 / Step^k) * \sum_i Coeff_i * f(x + Step * Loc_i).
+type Formula struct {
 	// Stencil is the set of sampling Points which are used to estimate the
 	// derivative. The locations will be scaled by Step and are relative to x.
 	Stencil []Point
 	Order   int     // The order of the approximated derivative.
-	Step    float64 // Default step size for the method.
+	Step    float64 // Default step size for the formula.
```

Comment:
> The optimal step size for forward difference for well-behaved functions is of the order of sqrt(eps).

Citation? The conclusion seems to be correct, I've just never heard a mathematical argument made that says that before and I'd be interested in reading it. 

Lets put off the default behavior to the next PR.

I do think Central is a good default. 1) It's a second order method, and so can get much better derivatives for the default step size. I'd rather have the default be to get good answers rather than fast ones. You know eventually we're going to get a bunch of questions/bugs about optimize failing to converge, and the cause will be users trying to converge too far with finite difference derivatives. Better derivatives by default will decrease that problem. 2) I like that it's symmetric. If there are some directional effects because of the function, I'd like them to show up all the time and not just half the time.